eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -344,7 +359,8 @@ data ImpTestState era = ImpTestState
   , impNativeScripts :: !(Map ScriptHash (NativeScript era))
   , impLastTick :: !SlotNo
   , impGlobals :: !Globals
-  , impEvents :: [SomeSTSEvent era]
+  , impRecordTxs :: Bool
+  , impEvents :: (Seq.Seq (Tx TopTx era), Seq.Seq (SomeSTSEvent era))
```

Comment:
I am not a big fan of sticking two totally unrelated concepts together into the same field named `Events`. In other words sequence of transactions has nothing to do with STS events, which is what this field was designed for.
Would it be hard to make it into a separate field? (FTR. we don't need to rely on the `MonadWriter` for this, in fact I'd rather avoid using it for recording transactions)

Reply:
Yes, separating it seems much better, and not having to use Writer is very freeing